server/users/users.js

```diff
@@ -15,6 +16,52 @@ const UserFriend = db.define('friends', {});
 // will add methods to User (ex. User.getFriends())
 User.belongsToMany(User, { as: 'Friends', through: UserFriend });
 
+User.getUserInfo = fbId => {
+  User.findOne({ where: { fbId } })
+    .then(user => {
+      const token = jwt.encode(user, 'secret');
+      const response = {
+        user,
+        token,
+      };
+      return response;
+    })
+    .catch(err => err);
+};
+
+User.getUserFriends = user => getFriends().catch(err => err);
+
+User.findOrCreate = profile => {
+  const name = profile.displayName;
+  const picture = profile.photos[0].value;
+  const fbId = {
+    fbId: profile.id,
+  };
+  User.findOne(fbId)
+    .then(match => {
+      // create user if there's no match
+      if (!match) {
+        const newUser = {
+          name,
+          fbId,
+          picture,
+        };
+        return User.create(newUser);
+      }
+      // if user already exists, update user entry in the database
+      const updatedInfo = {
+        name,
+        picture,
+      };
+      return match.update(updatedInfo)
+      .then(data => data)
```

Comment:
Don't need these then and catches--it'll return the result of the update automatically and line 60 will catch